src/test/suite/checkFirstSuggestionDecoration.test.ts

```diff
@@ -0,0 +1,125 @@
+import * as assert from "assert";
+import { beforeEach, test, afterEach, suite } from "mocha";
+import * as sinon from "sinon";
+import { reset } from "ts-mockito";
+import { EventEmitter } from "events";
+import * as vscode from "vscode";
+import { activate, getDocUri, sleep } from "./utils/helper";
+import {
+  isProcessReadyForTest,
+  readLineMock,
+  requestResponseItems,
+  stdinMock,
+  stdoutMock,
+} from "../../binary/mockedRunProcess";
+import { resetBinaryForTesting } from "../../binary/requests/requests";
+import {
+  acceptInline,
+  makeAChange,
+  mockAutocomplete,
+  mockGetState,
+  moveToActivePosition,
+  triggerInline,
+} from "./utils/completion.utils";
+import {
+  anAutocompleteResponse,
+  INLINE_NEW_PREFIX,
+  INLINE_PREFIX,
+  SINGLE_CHANGE_CHARACTER,
+} from "./utils/testData";
+import * as vscodeUtils from "../../vscode.api";
+import { firstSuggestionDecorationType } from "../../firstSuggestionDecoration";
+
+const nonEmptyArray = sinon.match((v) => Array.isArray(v) && v.length);
+
+suite("First suggestion decoration", () => {
```

Comment:
I had to give this file a weird name in order for it to run before the `fetchBinaryPath` test. Because all the tests run in the same extension host they may interfere with one another's environment, and `fetchBinaryPath` does something that makes this new test fail. I think we should put in place something like a super suite, that'll allow us to determine the order in which the tests will run manually, rather than having them just run in alphabetical order. @dimacodota

Reply:
In general, if tests are sensitive to execution order they're not done right. 
I think we should look into this issue and understand exactly what is the link between the 2 tests, and break it.

Reply:
Besides while I was learning the ins and outs of these (highly complicated!!) tests, I found that there's some tests that are very flaky in the CI. I had some issues particularly with the notification polling test but I couldn't pin it down.